src/range.rs

```diff
@@ -47,6 +43,16 @@ impl fmt::Debug for TextRange {
 
 /// Identity methods.
 impl TextRange {
+    /// Creates a new `TextRange` with given `start` and `end.
+    ///
+    /// # Panics
+    ///
+    /// Panics if `end < start`.
+    pub fn new(start: TextSize, end: TextSize) -> TextRange {
```

Comment:
The precise constructor scheme is TBBS after this PR, the `new` is just the simplest thing to enforce the invariant.